File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Dict, Optional
from datetime import datetime

class KPIData(BaseModel):
    total_income: float
    total_expenses: float
    net_balance: float
    savings_rate: float
    largest_expense: Optional[Dict]
    largest_income: Optional[Dict]

class CategoryBreakdown(BaseModel):
    category: str
    amount: float
    percentage: float
    transaction_count: int

class TrendData(BaseModel):
    date: str
    income: float
    expenses: float
    balance: float

class AnalyticsResponse(BaseModel):
    period_start: datetime
    period_end: datetime
    kpis: KPIData
    category_breakdown: List[CategoryBreakdown]
    trends: List[TrendData]
    insights: List[str]
    alerts: List[str]

File: frontend/src/types/analytics.ts
export interface KPIData {
  total_income: number;
  total_expenses: number;
  net_balance: number;
  savings_rate: number;
  largest_expense?: {
    description: string;
    amount: number;
    date: string;
  };
  largest_income?: {
    description: string;
    amount: number;
    date: string;
  };
}

export interface CategoryBreakdown {
  category: string;
  amount: number;
  percentage: number;
  transaction_count: number;
}

export interface TrendData {
  date: string;
  income: number;
  expenses: number;
  balance: number;
}

export interface AnalyticsData {
  period_start: string;
  period_end: string;
  kpis: KPIData;
  category_breakdown: CategoryBreakdown[];
  trends: TrendData[];
  insights: string[];
  alerts: string[];
}